core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala

```diff
@@ -103,6 +103,8 @@ trait IntegrationTestHarness extends KafkaServerTestHarness {
   override def tearDown() {
     producers.foreach(_.close())
     consumers.foreach(_.close())
+    producers.clear()
+    consumers.clear()
```

Comment:
I thought `setup()` and `tearDown()` will be executed once for every test method (e.g. testConsumeAfterPurge) but the initialization of `consumers` in `trait IntegrationTestHarness` will be executed only once per test class (e.g. AdminClientTest). I am wrong. I have reverted this change.